File: Python/program/Gpu.py
from Komponen import Komponen
from Connectable import Connectable

class Gpu(Komponen, Connectable):
    def __init__(self, merk, nama, vramGB, clockSpeedGHz, pcieVersion):
        super().__init__(merk, nama)
        self.vramGB = vramGB
        self.clockSpeedGHz = clockSpeedGHz
        self.pcieVersion = pcieVersion

    # Setter
    def setVramGB(self, vramGB):
        self.vramGB = vramGB

    def setClockSpeedGHZ(self, clockSpeedGHz):
        self.clockSpeedGHz = clockSpeedGHz

    def setPcieVersion(self, pcieVersion):
        self.pcieVersion = pcieVersion

    # Getter
    def getVramGB(self):
        return self.vramGB

    def getClockSpeedGHZ(self):
        return self.clockSpeedGHz

    def getPcieVersion(self):
        return self.pcieVersion

    # Implementasi dari Connectable
    def getConnectionType(self):
        return f"{self.pcieVersion}"

    def displayInfo(self):
        print(f"{self.getNama()} (VRAM: {self.vramGB} GB, Clock Speed: {self.clockSpeedGHz} GHz, PCIe Version: {self.pcieVersion})")

File: CPP/Program/Gpu.cpp
#include "Komponen.cpp"
#include "Connectable.cpp"

using namespace std;

class Gpu : public Komponen, public Connectable {
    private:
        int vramGB;
        float clockSpeedGHz;
        string pcieVersion;
    public:
        Gpu(){}

        Gpu(string merk, string nama, int vramGB, float clockSpeedGHz, string pcieVersion) 
            : Komponen(merk, nama) {
            this->vramGB = vramGB;
            this->clockSpeedGHz = clockSpeedGHz;
            this->pcieVersion = pcieVersion;
        }

        // setter
        void setVramGB(int vramGB){ this->vramGB = vramGB; }
        void setClockSpeedGHz(float clockSpeedGHz){ this->clockSpeedGHz = clockSpeedGHz; }
        void setPcieVersion(string pcieVersion){ this->pcieVersion = pcieVersion; }

        // getter
        int getVramGB(){ return vramGB; }
        float getClockSpeedGHz(){ return clockSpeedGHz; }
        string getPcieVersion(){ return pcieVersion; }

        // Implementasi dari Connectable
        string getConnectionType() override {
            return "PCIe Version: " + pcieVersion;
        }

        void displayInfo() override {   // Override method
            cout << getNama() << " (VRAM: " << vramGB << " GB, Clock Speed: " << clockSpeedGHz << " GHz, PCIe Version: " << pcieVersion << ")" << endl;
        }

        ~Gpu(){}
};
